### PR TITLE
DATA: Participant Model & Migration

### LINKED ISSUE #35: DATA: Participant Model & Migration

### PR DESCRIPTION
**Pull request by Nighila**
Closes #35

### CODE REVIEW

File: MumsWhoCode.Services.api/Brokers/Storages/StorageBroker.Participant.References.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            modelBuilder.Entity<Participant>()
+                .HasOne(Participant => Participant.CreatedByUser)
+                .WithMany(user => user.CreatedParticipants)
+                .HasForeignKey(category => category.CreatedBy)
+                .OnDelete(DeleteBehavior.NoAction);
+
+            modelBuilder.Entity<Participant>()
+                .HasOne(Participant => Participant.UpdatedByUser)
+                .WithMany(user => user.UpdatedParticipants)
+                .HasForeignKey(category => category.UpdatedBy)
```
Comment (NaynaMallick) [inline MumsWhoCode.Services.api/Brokers/Storages/StorageBroker.Participant.References.cs:23]: Change category to participant.

File: MumsWhoCode.Services.api/Brokers/Storages/StorageBroker.Participant.References.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+namespace MumsWhoCode.Services.Api.Brokers.Storages
+{
+    public partial class StorageBroker
+    {
+        private static void SetParticipantReferences(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<Participant>()
+                .HasOne(Participant => Participant.CreatedByUser)
+                .WithMany(user => user.CreatedParticipants)
+                .HasForeignKey(category => category.CreatedBy)
```
Comment (NaynaMallick) [inline MumsWhoCode.Services.api/Brokers/Storages/StorageBroker.Participant.References.cs:17]: Change category to participant.

File: MumsWhoCode.Services.api/Brokers/Storages/StorageBroker.Participant.References.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+using MumsWhoCode.Services.Api.Models.Participants;
+
+namespace MumsWhoCode.Services.Api.Brokers.Storages
+{
+    public partial class StorageBroker
+    {
+        private static void SetParticipantReferences(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<Participant>()
+                .HasOne(Participant => Participant.CreatedByUser)
```
Comment (ShriHumrudha) [inline MumsWhoCode.Services.api/Brokers/Storages/StorageBroker.Participant.References.cs:15]: small 'p'
and same for the below code

File: MumsWhoCode.Services.api/Models/Participants/Participant.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+using MumsWhoCode.Services.Api.Models.Users;
+
+namespace MumsWhoCode.Services.Api.Models.Participants
+{
+    public class Participant
+    {
+        public Guid Id { get; set; }
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public enum Gender
```
Comment (ShriHumrudha) [inline MumsWhoCode.Services.api/Models/Participants/Participant.cs:15]: Move the enums to separate files

File: MumsWhoCode.Services.api/Models/Participants/Participant.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        { 
+            Female,
+            Male,
+            Other 
+        }
+        public string mobile { get; set; }
```
Comment (ShriHumrudha) [inline MumsWhoCode.Services.api/Models/Participants/Participant.cs:21]: The properties name should be captialized

File: MumsWhoCode.Services.api/Models/Participants/Participant.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        }
+        public DateTimeOffset CreatedDate { get; set; }
+        public DateTimeOffset UpdatedDate { get; set; }
+
+        public Guid CreatedBy { get; set; }
+        public User CreatedByUser { get; set; }
+
+        public Guid UpdatedBy { get; set; }
+        public User UpdatedByUser { get; set; }
+
```
Comment (ShriHumrudha) [inline MumsWhoCode.Services.api/Models/Participants/Participant.cs:39]: Remove extra line

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

**Review by ShriHumrudha — APPROVED** [review]

**Review by NaynaMallick — APPROVED** [review]

File: MumsWhoCode.Services.api/Models/Participants/Participant.cs (hunk 5 above)
Comment (Nighila) [reply MumsWhoCode.Services.api/Models/Participants/Participant.cs]: resolved

**Pull request merged**